### ISSUE TITLE
add possibility to quit the com server

### ISSUE THREAD

**Issue by sdementen** [ericremoreynolds/excelpython#59]
Hello,

I need to be able to quit the COM server from VBA (for instance, when modules or PYTHONPATH is changed) in xlwings which uses excelpython COM server (ZoomerAnalytics/xlwings#246)

I have added the function in xlpyserver.py (or server.py in xlwings)

```
def Quit(self):
    global main_thread_id
    win32api.PostThreadMessage(main_thread_id, win32con.WM_QUIT, 0, 0)
```

with the main_thread_id a global variable initialised to win32api.GetCurrentThreadId().

It appears to work, ie the pythonw process is remove when executing Py.Quit

Does this looks right ?

If OK and useful, I can do a PR.

**Comment by ericremoreynolds** [comment]
Hi!

That does look pretty much right, please go ahead and make the PR!

Thanks very much.

Kind regards,

Eric

**Comment by sdementen** [comment]
Hello Eric,

I have tested the code on the xlwings version of the server and the pythonw process quits indeed.
However, when I restart the server in the same Excel instance by calling Py.Exec from VBA - I do this for instance when the PYTHONPATH is changed by the user - it does not work well (I can't remember exactly what happens but there are issues).

Is quitting the server enough to be able to restart a clean server at the next use of Py.Exec ?
Shouldn't Config::ActivateRPCServer also be adapted in some way ?
Should I free the library that was loaded after quitting the process (with FreeLibrary) ?

**Comment by ericremoreynolds** [comment]
Hi Sebastien,

I am not sure what issues you're encountering, however if the (correct) Python process is not running when `Py.Exec` (or any `Py.*` method) is invoked, then a new process is launched. Indeed if you change the `xlpython.cfg` file, for example changing the Python path specified in it, this should be enough to trigger a restart. Obviously this is not the case if only the environment variable is changed.

If you give me more precise details of what issues you're facing perhaps I can offer you more assistence.

Eric

**Comment by sdementen** [comment]
I'll come back with more details in a week or two (need to find time to
reproduce the bug). Sebastien

**Comment by fzumstein** [comment]
implemented in xlwings 0.11
